modules/imgproc/include/visp3/imgproc/vpCircleHoughTransform.h

```diff
@@ -0,0 +1,825 @@
+        params.m_minRadius = std::min(radiusLimits.first, radiusLimits.second);
+        params.m_maxRadius = std::max(radiusLimits.first, radiusLimits.second);
+
+        params.m_dilatationNbIter = j.value("dilatationNbIter", params.m_dilatationNbIter);
+
+        params.m_centerThresh = j.value("centerThresh", params.m_centerThresh);
+        if (params.m_centerThresh <= 0)
+        {
+          throw vpException(vpException::badValue, "Votes thresholds for center detection must be positive.");
+        }
+
+        params.m_radiusRatioThresh = j.value("radiusThreshRatio", params.m_radiusRatioThresh);
+
+        params.m_circlePerfectness = j.value("circlePerfectnessThreshold", params.m_circlePerfectness);
+
+        if (params.m_circlePerfectness <= 0 || params.m_circlePerfectness > 1)
+        {
+          throw vpException(vpException::badValue, "Circle perfectness must be in the interval ] 0; 1].");
+        }
+
+        params.m_centerMinDist = j.value("centerMinDistance", params.m_centerMinDist);
+        if (params.m_centerMinDist <= 0)
+        {
+          throw vpException(vpException::badValue, "Centers minimum distance threshold must be positive.");
+        }
+
+        params.m_mergingRadiusDiffThresh = j.value("mergingRadiusDiffThresh", params.m_mergingRadiusDiffThresh);
+        if (params.m_mergingRadiusDiffThresh <= 0)
+        {
+          throw vpException(vpException::badValue, "Radius difference merging threshold must be positive.");
+        }
+      }
+
+      /**
+       * \brief Parse a vpCircleHoughTransform into JSON format.
+       *
+       * \param j A JSON parser object.
+       * \param config The vpCircleHoughTransform that must be parsed into JSON format.
+       */
+      inline friend void to_json(json &j, const CHTransformParameters &params)
+      {
+        std::pair<unsigned int, unsigned int> radiusLimits = {params.m_minRadius, params.m_maxRadius};
+
+        j = json{
+            {"gaussianKernelSize", params.m_gaussianKernelSize},
+            {"gaussianStdev", params.m_gaussianStdev},
+            {"sobelKernelSize", params.m_sobelKernelSize},
+            {"cannyThresh", params.m_cannyThresh},
+            {"centerXlimits", params.m_centerXlimits},
+            {"centerYlimits", params.m_centerYlimits},
+            {"radiusLimits", radiusLimits},
+            {"dilatationNbIter", params.m_dilatationNbIter},
+            {"centerThresh", params.m_centerThresh},
+            {"radiusThreshRatio", params.m_radiusRatioThresh},
+            {"circlePerfectnessThreshold", params.m_circlePerfectness},
+            {"centerMinDistance", params.m_centerMinDist},
+            {"mergingRadiusDiffThresh", params.m_mergingRadiusDiffThresh}};
+      }
+    #endif
+  } ;
+
+  /**
+   * \brief Class that defines a 2D circle in an image.
+   */
+  class vpCircle2D
```

Comment:
`vpCircle2D` class should be moved in `module/core/include/visp3/vpCircle2D.h`
Since we have for a point in an image the `vpImagePoint` class, instead of `vpCircle2D` class name and corresponding files, I think that we should call the class `vpImageCircle` to be homogeneous with existing code

Reply:
The header of the `vpImageCircle.h` should have something similar to:

```
#include <visp3/core/config.h>
#if defined(VISP_HAVE_OPENCV>
#include <opencv2/core.hpp>
#endif

class VISP_EXPORT vpImageCircle